src/Ocelot/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -47,7 +50,9 @@ public static IOcelotBuilder AddOcelot(this IServiceCollection services, IConfig
     /// <returns>An <see cref="IOcelotBuilder"/> object.</returns>
     public static IOcelotBuilder AddOcelotUsingBuilder(this IServiceCollection services, Func<IMvcCoreBuilder, Assembly, IMvcCoreBuilder> customBuilder)
     {
-        var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+        var descriptor = services.Where(descriptor => descriptor.ServiceType == typeof(IConfiguration)).FirstOrDefault();
+        var provider = new ServiceCollection().Add(descriptor).BuildServiceProvider();
+        var configuration = provider.GetRequiredService<IConfiguration>();
```

Comment:
Why not to define a private helper to search for `IConfiguration` object?
Also, the XML description need to point out to the developer that the interface object is derived from DI and must be registered in DI container anyway!

Reply:
This is my fault when I reviewed #1655  and didn't pay my attention to the fact we add overloaded version without the interface object 👇 
https://github.com/ThreeMammals/Ocelot/blob/5a3c55e358afb7853819e3370daad9bb15452dd9/src/Ocelot/DependencyInjection/ServiceCollectionExtensions.cs#L48-L52
So, 😮  But Ocelot Builder has required 2nd argument:
https://github.com/ThreeMammals/Ocelot/blob/5a3c55e358afb7853819e3370daad9bb15452dd9/src/Ocelot/DependencyInjection/OcelotBuilder.cs#L51

Reply:
The fact that `OcelotBuilder` requires an `IConfiguration` is actually the problem in the first place.
This builder should only focus on registering service descriptors with their lifetime managers.
Services would access configuration by injection of `IOptionsMonitor<T>` rather than an IConfiguration reference.
Overloads with IConfiguration parameters would then be the ones to deprecate.

Reply:
I wasn't sure deduplication of 3 lines of code was worth writing a separate helper.
Also in my mind this code is expected to go away once the IConfiguration references are gone.

Reply:
@raman-m I've added some default behavior in case the IConfiguration wasn't registered.
Note that Tom's code reads the configuration from current working path, which may be different from the executable's path.